_posts/2021-11-23-smallrye-stork-intro.adoc

```diff
@@ -0,0 +1,68 @@
+
+Out of the box, it provides integrations with Hashicorp Consul, Eureka, and Kubernetes and brings in a couple of load-balancer (handling the service selection) implementations, such as round-robin. But the most significant strength of Stork is its extensibility. You can plug your own service discovery mechanism or implement your own, maybe business-related, service selection strategy.
+
+== Stork and Quarkus
+As of Quarkus 2.5.0.Final, Quarkus gRPC and the Quarkus Reactive REST Client can use Stork to handle the discovery and the selection of the services.
+
+You just need to configure how you want to locate and select your services and add the related implementation, and the rest will be handled by the clients directly.
+
+For example, to use Stork to determine the actual address for REST Client Reactive, set the scheme of your client's URI to `stork`, and the hostname of the URI to the name of the Stork service. If your client's `configKey` is `my-client`, to use it with Hashicorp Consul and the `round-robin` load balancer, put the following in your `application.properties:
+
+[source,property]
+.application.properties
+----
+quarkus.rest-client.my-client.uri=stork://hello-service
+
+stork.hello-service.service-discovery=consul
+stork.hello-service.service-discovery.consul-host=<consul host>
+stork.hello-service.service-discovery.consul-port=<consul port>
+
+stork.hello-service.load-balancer=round-robin
+----
+
+The http://smallrye.io/smallrye-stork/1.0.0.Beta1/quarkus/[project's documentation] describes step by step how to use Stork with the REST Client, using Consul and round-robin load balancer as examples.
+
+Service selection can be pretty advanced. For instance, the least-response-time strategy monitors the service calls to select the _fastest_ service instance.
+
+NOTE: In Quarkus 2.5, the gRPC extension does not propagate statistics of the calls, such as response time or encountered errors. Therefore, a load balancer that relies on this data, such as least-response-time, will not work correctly. We are working on it and hope to have it in Quarkus 2.6.
+
+
+=== Demo and a lot more!
+Take a look at the https://www.youtube.com/watch?v=l3mLKU3wR2A[70th episode of Quarkus Insights] for a more detailed explanation and examples. The demo used https://github.com/michalszynkiewicz/rest-client-reactive-stork[this project] as the client-side, each of the steps has a corresponding commit in the main branch. The backend services were instances of https://github.com/michalszynkiewicz/configurable-rest-service[this project], and the custom service discovery server implementation can be found https://github.com/michalszynkiewicz/simple-service-discovery-server[here].
+
+== Conclusion
+
+This post briefly introduces SmallRye Stork and its Quarkus integration. Stork provides a way to locate and select services. It's simple and customizable.
+We will cover Stork features in more detail shortly.
```

Comment:
In more details

Reply:
nope, it's "in more detail" according to all pages on English I found ;)